src/components/Card.js

```diff
@@ -1,56 +1,179 @@
+  //   if (isLiked) {
+  //     api.dislikeACard(cardId).then(() => {
+  //       if (Array.isArray(data.likes)) {
+  //         this._updateLikes(data.likes);
+  //       } else {
+  //         console.error("Invalid data format for likes:", !isLiked);
+  //       }
+  //     });
+  //   } else {
+  //     api.likeACard(cardId).then((data) => {
+  //       if (Array.isArray(data.likes)) {
+  //         this._updateLikes(data.likes);
+  //       } else {
+  //         console.error("Invalid data format for likes:", !isLiked);
+  //       }
+  //     });
+  //   }
+  // }
+
+  // _handleLikeClick(cardId, isLiked) {
+  //   cardId._toggleLikeButton();
+  // }
+
+  // _updateLikes(likes) {
+  //   this._likes = likes;
+  //   this._cardElement.querySelector(".card__like-count").textContent =
+  //     likes.length;
+  //   this._cardElement
+  //     .querySelector(".card__like-button")
+  //     .classList.toggle(
+  //       "card__like-button_active",
+  //       this._likes.includes(this._id)
+  //     );
+  //   this._handleLikeIcon();
+  // }
+
+  // turn into public method and call it after succesful delete request
+  handleDeleteCard() {
     this._cardElement.remove();
     this._cardElement = null;
   }
 
+  // getView() {
+  //   this._cardElement = document
+  //     .querySelector(this._cardSelector)
+  //     .content.querySelector(".card")
+  //     .cloneNode(true);
+  //   // TODO insert the src, alt text, and text content - DONE
+  //   this._cardImageEl = this._cardElement.querySelector(".card__image");
+  //   this._cardImageEl.src = this._link;
+  //   this._cardImageEl.alt = this._name;
+  //   this._cardTitleEl = this._cardElement.querySelector(".card__title");
+  //   this._cardTitleEl.textContent = this._name;
+
+  //   // set event listeners,
+  //   this._setEventListeners();
+  //   // TODO return the card
+  //   return this._cardElement;
+  // }
+
+  _getTemplate() {
+    const templateElement = document.querySelector(this._cardSelector);
+    if (!templateElement) {
+      throw new Error(`Template selector "${this._cardSelector}" not found.`);
+    }
+    return templateElement.content.querySelector(".card").cloneNode(true);
+  }
+
   getView() {
-    this._cardElement = document
-      .querySelector(this._cardSelector)
-      .content.querySelector(".card")
-      .cloneNode(true);
-    // TODO insert the src, alt text, and text content - DONE
-    this._cardImageEl = this._cardElement.querySelector(".card__image");
-    this._cardImageEl.src = this._link;
-    this._cardImageEl.alt = this._name;
-    this._cardTitleEl = this._cardElement.querySelector(".card__title");
-    this._cardTitleEl.textContent = this._name;
-
-    // set event listeners,
+    this._cardElement = this._getTemplate();
+    this._cardElement.setAttribute("data-id", this._id);
     this._setEventListeners();
-    // TODO return the card
+    this._cardElement.querySelector(".card__image").src = this._link;
```

Comment:
Every tag `img` should have an informative attribute `alt` (the **name** of the image is usually there) to show the `alternative` text if the image is not uploaded, and it’s needed for **screen readers**